File: Payroll/Payroll/Payroll/Controllers/ErpController.cs
using DataLayer;
using Payroll_HCC.Infrastructure;
using System;
using System.Web.Mvc;

namespace Payroll_HCC.Controllers
{
    /// <summary>
    /// ERP platform endpoints (same contract as LIMS' ErpController):
    ///  GET  /erp/touch  - pre-authenticates this app from the erp_sso cookie (called by the Applications Library).
    ///  POST /erp/verify - lets the central login verify a Payroll-native account (X-Erp-Secret shared secret).
    /// </summary>
    public class ErpController : Controller
    {
        [HttpGet]
        public ActionResult Touch()
        {
            Response.Cache.SetCacheability(System.Web.HttpCacheability.NoCache);
            Response.Cache.SetNoStore();
            if (!ErpSso.Enabled) return new HttpStatusCodeResult(204);
            if (App.CurrentUser != null) return new HttpStatusCodeResult(204);
            string token = ErpSso.CookieToken(Request);
            if (token == null) return new HttpStatusCodeResult(204);
            try
            {
                var id = ErpSso.Validate(token);
                if (id != null) ErpSso.SignIn(HttpContext, token, id);
            }
            catch (Exception ex) { FileLogger.Error("erp/touch failed.", ex); }
            return new HttpStatusCodeResult(204);
        }

        [HttpPost]
        public ActionResult Verify(string username, string password)
        {
            string secret = Request.Headers["X-Erp-Secret"];
            if (string.IsNullOrEmpty(ErpSso.SharedSecret) || secret != ErpSso.SharedSecret)
                return new HttpStatusCodeResult(403);
            string reason;
            SessionUser u = App.Security.Authenticate(username, password, out reason);
            if (u == null) return Json(new { ok = false });
            return Json(new { ok = true, id = u.id, email = u.Email, name = u.DisplayName, isAdmin = u.IsAdministrator });
        }
    }
}

File: LMIS/app/Http/Controllers/ErpController.php
<?php

namespace App\Http\Controllers;

use App\Models\User;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Hash;

/**
 * ERP platform endpoints used by the PMS host (the single login).
 *
 *  GET  /erp/touch   – "pre-authenticate": runs through ErpSso like any page and reports the
 *                      signed-in state. /Apps calls it right after login so LIMS is already
 *                      signed in before the user clicks its tile.
 *  POST /erp/verify  – credential check for LIMS-native accounts, so the ONE login page accepts
 *                      every solution's credentials. Only callable by the PMS host with the
 *                      shared secret; never returns the hash; rate limited.
 */
class ErpController extends Controller
{
    public function touch(Request $request)
    {
        return response()->json([
            'app'           => config('erp.app_code'),
            'authenticated' => Auth::check(),
            'user'          => Auth::check() ? ['id' => Auth::id(), 'name' => Auth::user()->name, 'email' => Auth::user()->email] : null,
        ], 200, ['Cache-Control' => 'no-store']);
    }

    public function verify(Request $request)
    {
        $secret = (string) config('erp.shared_secret');
        if ($secret === '' || !hash_equals($secret, (string) $request->header('X-Erp-Secret', ''))) {
            return response()->json(['ok' => false, 'error' => 'forbidden'], 403);
        }
        $username = trim((string) $request->input('username', ''));
        $password = (string) $request->input('password', '');
        if ($username === '' || $password === '') {
            return response()->json(['ok' => false], 200);
        }
        $user = User::where('email', $username)->orderBy('id')->first();
        if (!$user || !Hash::check($password, $user->password)) {
            return response()->json(['ok' => false], 200);
        }
        if ((int) ($user->isDeleted ?? 0) === 1) {
            return response()->json(['ok' => false], 200);
        }
        return response()->json([
            'ok'   => true,
            'user' => ['id' => $user->id, 'email' => $user->email, 'name' => $user->name, 'is_admin' => (int) ($user->is_admin ?? 0)],
        ]);
    }
}
